web-app/src/components/UserModal/index.js

```diff
@@ -1,67 +1,42 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { PropTypes as PT } from 'prop-types';
+import container from './container';
 import { Modal, Button, Email } from '../../components/common';
 import { StyleContainer, Stat, StatWrap } from './styled';
-import { getHolidays } from '../../services/holidayService';
-import swal from 'sweetalert2';
+import { isEmpty } from 'lodash';
 import { getTotalDaysInEventArrayWithStatus } from '../../utilities/dates';
 import HolidayStatus from '../../utilities/holidayStatus';
-import { isEmpty } from 'lodash';
-
-class UserModal extends Component {
-  static propTypes = {
-    user: PT.object,
-    closeModal: PT.func.isRequired,
-    history: PT.object.isRequired,
-  };
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      userHolidays: [],
-    };
-  }
 
-  componentDidMount() {
-    const { user } = this.props;
-    if (isEmpty(user)) return null;
+const UserModal = ({
+  user,
+  closeModal, 
+  history,
+  userHolidays,
+  hasPermission,
+}) => {
 
-    getHolidays(user.employeeId)
-      .then(response => {
-        this.setState({ userHolidays: response.data });
-      })
-      .catch(error => {
-        swal(
-          'Error',
-          `Error getting user holiday details: ${error.message}`,
-          'error'
-        );
-      });
-  }
-
-  getTotalPendingDays = () => {
```

Comment:
Saying you've built a container it might make sense to move `getTotalPendingDays` and `getTotalApprovedDays` into the container and pass down props `approvedDays` and `pendingDays` instead.